generic_command.py

```diff
@@ -0,0 +1,34 @@
+import asyncio
+
+import iterm2
+
+
+SCRIPT = 'SCRIPT'
+PREFIX = 'PREFIX'
+
+async def main(connection):
+    component = iterm2.StatusBarComponent(
+        short_description='Execute a user defined command',
+        detailed_description='A super simple component where both the command creating the text and a prefix can be defined by the user',
+        exemplar='⎈ docker-for-desktop',
+        update_cadence=2,
+        identifier='engineering.dane.iterm-components.generic-command',
+        knobs=[
+           iterm2.StringKnob('Script', 'echo "hello world"', '', SCRIPT),
+           iterm2.StringKnob('Prefix', '>', '', PREFIX)
+        ],
+    )
+
+    @iterm2.StatusBarRPC
+    async def generic_command_coroutine(knobs):
+        proc = await asyncio.create_subprocess_shell(
+            knobs[SCRIPT],
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+        stdout, stderr = await proc.communicate()
+        return f'{knobs[PREFIX]} {stdout.decode().strip()}' if not stderr else 'Command failed!'
```

Comment:
This leaves an extra space in front of the output if there isn't a prefix defined. Can fix it up with the following:

```suggestion
        prefix = knobs[PREFIX]
        prefix = f'{prefix} ' if prefix else ''
        return f'{prefix}{stdout.decode().strip()}' if not stderr else 'Command failed!'
```